src/reducers/provider-state-updaters.js

```diff
@@ -0,0 +1,200 @@
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {withTask} from 'react-palm/tasks';
+import {default as Console} from 'global/console';
+import {generateHashId, getError} from 'utils/utils';
+import {EXPORT_FILE_TO_CLOUD_TASK, ACTION_TASK, DELAY_TASK} from 'tasks/tasks';
+import {exportFileSuccess, exportFileError, saveToCloudSuccess} from 'actions/provider-actions';
+import {
+  removeNotification,
+  toggleModal,
+  addNotification
+} from 'actions/ui-state-actions';
+import {
+  DEFAULT_NOTIFICATION_TYPES,
+  DEFAULT_NOTIFICATION_TOPICS
+} from 'constants/default-settings';
+
+export const INITIAL_PROVIDER_STATE = {
+  isLoading: false,
+  error: null,
+  currentProvider: null,
+  successInfo: {}
+};
+
+function createFileJson(mapData) {
+  const {map, info} = mapData;
+  const data = JSON.stringify(map);
+  const newBlob = new Blob([data], {type: 'application/json'});
+
+  // TODO: Allow user import file name
+  const name =
+    info.title && info.title.length
+      ? info.title
+      : `keplergl_${generateHashId(6)}`;
+  const fileName = `${name}.json`;
+
+  return {file: new File([newBlob], fileName), fileName};
+}
+
+function createActionTask(action, payload) {
+  if (typeof action === 'function') {
+    return ACTION_TASK().map(_ => action(payload));
+  }
+
+  return null;
+}
+
+/**
+ * This method will export the current kepler config file to the chosen cloud proder
+ * add returns a share URL
+ *
+ * @param {*} state
+ * @param {*} action
+ */
+export const exportFileToCloudUpdater = (state, action) => {
+  const {mapData, provider, isPublic, onSuccess, onError, closeModal} = action.payload;
+  console.log(mapData);
+
+  if (!provider) {
+    Console.error(`provider is not defined`);
+    return state;
+  }
+
+  if (typeof provider.uploadFile !== 'function') {
+    Console.error(
+      `uploadFile is not a function of Cloud provider: ${provider.name}`
+    );
+    return state;
+  }
+
+  // TODO: do we need to always create a fileBlob?
```

Comment:
I double checked the API (https://dropbox.github.io/dropbox-sdk-js/Dropbox.html#filesUpload) and based on the type accepted by the API we may be able to pass any object: https://dropbox.github.io/dropbox-sdk-js/global.html#FilesCommitInfo

Reply:
I agree the provider should take care of handling the json the way it wants